tailcfg/tailcfg.go

```diff
@@ -2653,6 +2653,17 @@ const (
 	// NodeAttrTrafficSteering configures the node to use the traffic
 	// steering subsystem for via routes. See tailscale/corp#29966.
 	NodeAttrTrafficSteering NodeCapability = "traffic-steering"
+
+	// NodeAttrTailnetDisplayName is how the tailnet name should be displayed to the user.
+	// Currently if this is set by a user this will only include letters, numbers,
+	// apostrophe, spaces, and hyphens. The only time this is not true,
+	// is if it defaults to the SID.
+	// i.e. "Tailscale - Dev", "User's Tailnet", or if it is a default SID
+	// "user@example.com"
+	// If a user manually set a display name the value that will be set here.
+	// If a user has not manually set a display name or has removed a display name
+	// then this value will be populated with the TailnetSID.
```

Comment:
"TailnetSID" isn't a concept that's used in this open source repo. I see only one other reference but that other one's kinda okay because it's only in parens after other examples.

We can shorten this to something like:

    // NodeAttrTailnetDisplayName is an optional alternate name for the tailnet to be displayed to the user.
    // If empty or absent, a default is used.
    // Values can look like "foo.com" or "Foo's Test Tailnet - Staging".